cloud/parameterstore/parameter_manager.go

```diff
@@ -0,0 +1,196 @@
+package parameterstore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmTypes "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// Parameter represents a parameter kept in Parameter Store.
+type Parameter struct {
+	// Name is the full name of the parameter, including its path. This is a
+	// unique identifier. For example, the full name could be
+	// /evergreen/path/to/my-parameter.
+	Name string `bson:"-" json:"-" yaml:"-"`
```

Comment:
Would it make sense to store just the path here so that it is more Extensible? Perhaps with a Name() function that returns the full name by combining the two?

Reply:
I don't think there's really a use case to store the path without the basename. We can add it later if there's a use for it, but there's no actual use case for it currently.

Reply:
It is harder to transition over once we've started using it than it is to store it that way from the start.

Reply:
It doesn't seem like that? Since this is returned as a struct, it's always easy to extend it to add more fields as needed (removing/modifying is the only concern since that requires fixing). I intentionally returned a struct because it would be easy to extend later on and return additional information.

Reply:
Ah that's true, since this is not stored in our db it is easier to change after.